tests/foreman/performance/common/perf_concurrent_test.py

```diff
@@ -0,0 +1,372 @@
+                               current_num_threads, is_attach):
+        """Write stat of per-client results to csv file
+
+        note: take the full list of a client i; calculate stat results
+        """
+        for i in range(current_num_threads):
+            if is_attach:
+                time_list = time_result_dict.get('thread-{}'.format(i))[2]
+            else:
+                time_list = time_result_dict.get('thread-{}'.format(i))
+            threadName = 'client-{}'.format(i)
+            generate_stat_for_concurrent_thread(threadName, time_list,
+                                                stat_file_name,
+                                                len(time_list), 1)
+
+    def _write_stat_per_test(self, stat_file_name,
+                             time_result_dict, is_attach):
+        """Write stat of per-test results to csv file
+
+        note: take the full dictionary of test and calculate general results
+        """
+        full_list = []
+        for i in range(len(time_result_dict)):
+            if is_attach:
+                time_list = time_result_dict.get('thread-{}'.format(i))[2]
+            else:
+                time_list = time_result_dict.get('thread-{}'.format(i))
+            full_list += time_list
+
+        generate_stat_for_concurrent_thread('test-{}'
+                                            .format(len(time_result_dict)),
+                                            full_list, stat_file_name,
+                                            len(full_list), 1)
+
+    def kick_off_ak_test(self, current_num_threads, total_iterations):
+        """Refactor out concurrent register by ak test case
+
+        @para current_num_threads: number of threads
+        @para total_iterations
+        """
+        # check if number of threads are mapped with number of vms
+        current_vm_list = self.vm_list[:current_num_threads]
+        self.assertEqual(len(current_vm_list), current_num_threads)
+
+        # Parameter for statistics files
+        self._set_num_iterations(total_iterations, current_num_threads)
+        self._set_bucket_size()
+
+        # Create a list to store all threads
+        thread_list = []
+        # Create a dictionary to store all timing results from each client
+        time_result_dict = {}
+
+        # Create new threads and start each thread mapped with a vm
+        for i in range(current_num_threads):
+            threadName = "thread-{}".format(i)
+            time_result_dict[threadName] = []
+            thread = SubscribeAKPerfThread(
+                i, threadName, time_result_dict,
+                self.num_iterations, self.ak_name,
+                self.default_org, current_vm_list[i]
+            )
+            thread.start()
+            thread_list.append(thread)
+
+        # wait all threads in thread list
+        self._join_all_threads(thread_list)
+
+        # write raw result of ak
+        self._write_raw_csv_file(self.raw_file_name, time_result_dict,
```

Comment:
Place one argument on its own line:

```python
self._write_raw_csv_file(
    self.raw_file_name,
    time_result_dict,
    current_num_threads,
    'raw-ak-{}-clients'.format(current_num_threads)
)
```

This applies for other parts of the code and will help improve readability.